File: 1524_number_of_sub_arrays_with_odd_sum.java
class Solution {
    public int numOfSubarrays(int[] g) {
        long v = 0, z = 0, w = 1, mod = 1_000_000_007;
        for (int y : g) {
            z += y;
            v += z % 2;
            w += (z % 2 == 0) ? 1 : 0;
        }
        return (int) ((v * w) % mod);
    }
}

File: 1524_number_of_sub_arrays_with_odd_sum.py
class Solution(object):
    def numOfSubarrays(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        v = z = 0
        w = 1
        mod = 1_000_000_007

        for y in arr:
            z += y
            v += z % 2
            w += 1 if z % 2 == 0 else 0

        return (v * w) % mod
